### PR TITLE
backupccl: switch CREATE SCHEDULE to use the planner txn

### LINKED ISSUE #53341: scheduled_jobs: CREATE SCHEDULE FOR BACKUP should be transactional
When creating a scheduled backup in a transaction, if the transaction is rolledback, the schedule is still created:

```
root@:26257/defaultdb> show schedules;
  id | label | schedule_status | next_run | state | recurrence | jobsrunning | owner | created | command
-----+-------+-----------------+----------+-------+------------+-------------+-------+---------+----------
(0 rows)

Time: 6.081952ms

root@:26257/defaultdb> begin;
BEGIN

Time: 236.719µs

root@:26257/defaultdb  OPEN> CREATE SCHEDULE FOR BACKUP INTO 'nodelocal://0/backup?AWS_SECRET_ACCESS_KEY=neverappears' RECURRING '@hourly';
     schedule_id     |       name        | status |         first_run         | schedule |                                     backup_stmt
---------------------+-------------------+--------+---------------------------+----------+--------------------------------------------------------------------------------------
  583989888279609345 | BACKUP 1598289970 | ACTIVE | 2020-08-24 18:00:00+00:00 | @hourly  | BACKUP INTO 'nodelocal://0/backup?AWS_SECRET_ACCESS_KEY=neverappears' WITH detached
(1 row)

Time: 32.741536ms

root@:26257/defaultdb  OPEN> rollback;
ROLLBACK

Time: 4.804817ms

root@:26257/defaultdb> show schedules;
          id         |       label       | schedule_status |         next_run          | state | recurrence | jobsrunning | owner |             created              |                                                 command
---------------------+-------------------+-----------------+---------------------------+-------+------------+-------------+-------+----------------------------------+----------------------------------------------------------------------------------------------------------
  583989888279609345 | BACKUP 1598289970 | ACTIVE          | 2020-08-24 18:00:00+00:00 | NULL  | @hourly    |           0 | root  | 2020-08-24 17:26:10.395737+00:00 | {"backup_statement": "BACKUP INTO 'nodelocal://0/backup?AWS_SECRET_ACCESS_KEY=redacted' WITH detached"}
(1 row)

Time: 3.690144ms
```

**Expectations**
That the schedules created in a rolledback transaction are not created.

** Version **
```
Build Tag:        v20.2.0-alpha.3-471-g810eba5adf
Build Time:       2020/08/24 16:14:07
Distribution:     CCL
Platform:         linux amd64 (x86_64-linux-gnu)
Go Version:       go1.14.4
C Compiler:       gcc 7.5.0
Build Commit ID:  810eba5adf6552647e1b117cb8713932406d5e2f
Build Type:       development
```

### PR DESCRIPTION
**Pull request by adityamaru**
Fixes: #53341

Release justification: Bug fixes and low-risk updates to new
functionality

### CODE REVIEW

File: pkg/ccl/backupccl/create_scheduled_backup.go [hunk 1]
```diff
@@ -296,7 +295,8 @@ func doCreateBackupSchedules(
 	}
 
 	ex := p.ExecCfg().InternalExecutor
-	return p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+	createSchedules := func() error {
```
Comment (pbardea) [inline pkg/ccl/backupccl/create_scheduled_backup.go:298]: Do we need to define this as a method? Can we inline it?

**Review by pbardea — APPROVED** [review]
LGTM mod 1 nit

**Review by dt — APPROVED** [review]

**Comment by adityamaru** [comment]
TFTRs!

bors r+

**Comment by miretskiy** [comment]
nice :)

**Comment by adityamaru** [comment]
Retrying.

bors r+

**Comment by otan** [comment]
bors r=dt,pbardea

atoning for bors sins

**Pull request merged**